test/api/route/messages_test.dart

```diff
@@ -314,6 +314,20 @@ void main() {
 
     test('to stream', () {
       return FakeApiConnection.with_((connection) async {
+        await checkSendMessage(connection,
+          destination: StreamDestination(streamId, topic), content: content,
+          expectedBodyFields: {
+            'type': 'stream',
+            'to': streamId.toString(),
+            'topic': topic,
+            'content': content,
+            'read_by_sender': 'true',
+          });
+      });
+    });
+
+    test('to server without read_by_sender support', () {
```

Comment:
Conveniently, the Zulip server actually ignores extra parameters sent in a request. So we can simplify by always sending this parameter, without checking whether the server is new enough to understand it.